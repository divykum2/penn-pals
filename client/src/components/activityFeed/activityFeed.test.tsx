import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from "react-router-dom";
import ActivityFeed from './activityFeed';

test('renders activity feed', async() => {
  render(<Router><ActivityFeed /></Router>);

  expect(screen.getByText('Notifications')).toBeInTheDocument();
  expect(screen.getByText('Profile')).toBeInTheDocument();
  expect(screen.getByText('Logout')).toBeInTheDocument();

});

test('testing navigation', () => {
  render(<Router><ActivityFeed /></Router>);

  const button = screen.getByRole('button', {name: "Profile"});
  fireEvent.click(button);

  // expect(screen.getByText('Suggested Users')).toBeInTheDocument();

})

test('renders app logo', () => {
    render(<Router><ActivityFeed/></Router>);

    const image = screen.getByAltText('logo');
    expect(image).toHaveAttribute('src', 'logo.png');
})

test('Activity feed matches snapshot', () => {
    const component = renderer.create(<Router><ActivityFeed /></Router>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
