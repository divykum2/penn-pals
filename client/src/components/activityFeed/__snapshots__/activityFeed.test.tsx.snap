// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Activity feed matches snapshot 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-12n0uxm-MuiGrid-root"
>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
  >
    <div
      className="MuiBox-root css-k008qs"
    >
      <nav
        className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-11b3ww9-MuiPaper-root-MuiAppBar-root"
        style={
          Object {
            "background": "white",
            "opacity": "20",
          }
        }
      >
        <div
          className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
        >
          <div
            className="MuiTypography-root MuiTypography-h5 css-1o4wp2u-MuiTypography-root"
          >
            <img
              alt="logo"
              height="40"
              src="logo.png"
            />
          </div>
          <div
            className="MuiBox-root css-130f8nx"
          >
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-hrbc3j-MuiButtonBase-root-MuiButton-root"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              Notifications
            </button>
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-hrbc3j-MuiButtonBase-root-MuiButton-root"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              Profile
            </button>
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-hrbc3j-MuiButtonBase-root-MuiButton-root"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              Logout
            </button>
          </div>
        </div>
      </nav>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5 css-k3f0g4-MuiGrid-root"
    style={
      Object {
        "marginTop": "5%",
      }
    }
  >
    <div
      className="MuiBox-root css-0"
    >
      <div
        className="timeline"
      >
        <div
          className="post"
        >
          <div
            className="post__header"
          >
            <div
              className="MuiAvatar-root MuiAvatar-circular post__avatar css-1wlk0hk-MuiAvatar-root"
            >
              <img
                alt="ava"
                className="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
                src="/static/images/avatar/1.jpg"
              />
            </div>
            <h3
              onClick={[Function]}
              style={
                Object {
                  "cursor": "pointer",
                }
              }
            />
            <span
              style={
                Object {
                  "marginLeft": "25%",
                  "position": "absolute",
                }
              }
            >
              <svg
                aria-label="heartFill"
                fill="currentColor"
                height="1em"
                onClick={[Function]}
                stroke="currentColor"
                strokeWidth="0"
                style={
                  Object {
                    "color": undefined,
                    "cursor": "pointer",
                  }
                }
                viewBox="0 0 16 16"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M8 1.314C12.438-3.248 23.534 4.735 8 15-7.534 4.736 3.562-3.248 8 1.314z"
                  fillRule="evenodd"
                />
              </svg>
            </span>
            <svg
              aria-label="comment"
              fill="currentColor"
              height="1em"
              onClick={[Function]}
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                  "cursor": "pointer",
                  "marginLeft": "28%",
                  "position": "absolute",
                }
              }
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M20 2H4c-1.103 0-2 .897-2 2v18l5.333-4H20c1.103 0 2-.897 2-2V4c0-1.103-.897-2-2-2zm0 14H6.667L4 18V4h16v12z"
              />
            </svg>
          </div>
          <img
            alt="post"
            className="post__image"
            src=""
          />
          <h4
            className="post__text"
          >
            <strong
              onClick={[Function]}
              style={
                Object {
                  "cursor": "pointer",
                }
              }
            />
             
          </h4>
        </div>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1equabv-MuiGrid-root"
  >
    <div>
      <form
        style={
          Object {
            "marginTop": "27%",
          }
        }
      >
        <div
          className="MuiFormControl-root MuiTextField-root text css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
          onInput={[Function]}
          style={
            Object {
              "width": "250px",
            }
          }
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined css-1pysi21-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink={false}
            htmlFor="search-bar"
            id="search-bar-label"
          >
            Search a user
          </label>
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
              disabled={false}
              id="search-bar"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
            />
            <fieldset
              aria-hidden={true}
              className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="css-yjsfm1"
              >
                <span>
                  Search a user
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <button
          aria-label="search"
          className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
          disabled={false}
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="submit"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="SearchIcon"
            focusable="false"
            style={
              Object {
                "fill": "#171F57",
              }
            }
            viewBox="0 0 24 24"
          >
            <path
              d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
            />
          </svg>
        </button>
      </form>
      <div
        className="suggestedUser"
      >
        <div
          className="line"
        />
        <div
          style={
            Object {
              "color": "#7F190E",
              "fontSize": 20,
              "paddingLeft": "10px",
              "paddingTop": "10px",
              "textAlign": "center",
            }
          }
        >
          Suggested Users
        </div>
        <div
          className="userBlock"
        >
          <div
            className="user"
          >
            <div
              className="MuiAvatar-root MuiAvatar-circular post__avatar css-1wlk0hk-MuiAvatar-root"
              style={
                Object {
                  "display": "inline-flex",
                  "height": "40px",
                  "width": "40px",
                }
              }
            >
              <img
                className="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
                src="/static/images/avatar/1.jpg"
              />
            </div>
            <span
              className="text"
              onClick={[Function]}
              style={
                Object {
                  "borderWidth": "10px",
                  "color": "#7F190E",
                  "cursor": "pointer",
                  "fontSize": 18,
                  "justifyContent": "normal",
                  "margin": 1,
                  "width": "200px",
                }
              }
            >
              No More User To Follow
            </span>
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              Follow
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
